Tests/PubNubTests/Networking/Routers/PresenceRouterTests.swift

```diff
@@ -29,38 +29,42 @@
 import XCTest
 
 // swiftlint:disable:next type_body_length
-final class PresenceEndpointTests: XCTestCase {
+final class PresenceRouterTests: XCTestCase {
   var pubnub: PubNub!
   let config = PubNubConfiguration(publishKey: "FakeTestString", subscribeKey: "FakeTestString")
 
   let channelName = "TestChannel"
   let otherChannel = "OtherTestChannel"
+}
 
-  // MARK: - HereNow Tests
+// MARK: - HereNow Tests
 
-  func testHereNow_Endpoint() {
-    let endpoint = Endpoint.hereNow(channels: [channelName], groups: [], includeUUIDs: true, includeState: true)
+extension PresenceRouterTests {
+  func testHereNow_Router() {
+    let router = PresenceRouter(.hereNow(channels: [channelName], groups: [], includeUUIDs: true, includeState: true), configuration: config)
 
-    XCTAssertEqual(endpoint.description, "Here Now")
-    XCTAssertEqual(endpoint.category, .hereNow)
-    XCTAssertEqual(endpoint.operationCategory, .presence)
-    XCTAssertNil(endpoint.validationError)
+    XCTAssertEqual(router.endpoint.description, "Here Now")
+    XCTAssertEqual(router.category, "Here Now")
+    XCTAssertEqual(router.service, .presence)
   }
 
-  func testHereNow_Endpoint_ValidationError() {
-    let endpoint = Endpoint.hereNow(channels: [], groups: [], includeUUIDs: true, includeState: true)
+  func testHereNow_Router_ValidationError() {
+    let router = PresenceRouter(.hereNow(channels: [], groups: [], includeUUIDs: true, includeState: true), configuration: config)
 
-    XCTAssertNotEqual(endpoint.validationError?.pubNubError,
-                      PubNubError(.invalidEndpointType, endpoint: endpoint))
+    XCTAssertEqual(router.validationError?.pubNubError?.details.first,
+                      ErrorDescription.missingChannelsAnyGroups)
   }
 
-  func testHereNow_Endpoint_AssociatedValues() {
-    let endpoint = Endpoint.hereNow(channels: [], groups: [], includeUUIDs: true, includeState: true)
+  func testHereNow_Router_Channels() {
+    let router = PresenceRouter(.hereNow(channels: [channelName], groups: [], includeUUIDs: true, includeState: true), configuration: config)
 
-    XCTAssertEqual(endpoint.associatedValue["channels"] as? [String], [])
-    XCTAssertEqual(endpoint.associatedValue["groups"] as? [String], [])
-    XCTAssertEqual(endpoint.associatedValue["includeUUIDs"] as? Bool, true)
-    XCTAssertEqual(endpoint.associatedValue["includeState"] as? Bool, true)
+    XCTAssertEqual(router.endpoint.channels, [channelName])
+  }
+
+  func testHereNow_Router_Groups() {
+    let router = PresenceRouter(.hereNow(channels: [], groups: [channelName], includeUUIDs: true, includeState: true), configuration: config)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line should be 120 characters or less: currently 141 characters](https://app.codacy.com/app/PubNub/swift/pullRequest?prid=4513555)